simvue/run.py

```diff
@@ -982,17 +983,70 @@ def log_metrics(
     @check_run_initialised
     @skip_if_failed("_aborted", "_suppress_errors", False)
     @pydantic.validate_call
-    def save(
+    def save_object(
         self,
-        filename: str,
+        obj: typing.Any,
+        category: typing.Literal["input", "output", "code"],
+        name: typing.Optional[str] = None,
+        allow_pickle: bool = False,
+    ) -> bool:
+        obj: DeserializedContent
```

Comment:
Why is this typed here instead of within the function definition? Is it because pydantic can't handle it?

Reply:
Yes it is because `pydantic` does not like it, but to be honest this should serialize anything anyway so I will change it